data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/Pipeline.java

```diff
@@ -257,7 +257,7 @@ private void shutdownExecutorService(final ExecutorService executorService, fina
     List<Future<Void>> publishToSinks(final Collection<Record> records) {
         final int sinksSize = sinks.size();
         final List<Future<Void>> sinkFutures = new ArrayList<>(sinksSize);
-        final RouterGetRecordStrategy getRecordStrategy = new RouterCopyRecordStrategy(sinks);
+        final RouterGetRecordStrategy getRecordStrategy = new RouterCopyRecordStrategy(router.getEventFactory(), router.getAcknowledgementSetManager(), sinks);
```

Comment:
I think getting these into the `Pipeline` and then providing here is more appropriate.

Then `Router` would not need these values at all. The `Router` doesn't deal with creation or acknowledgment - it just moves data around.